### PR TITLE
[plugins/digitalocean][feat] Digital Ocean collector, part 1/2

### PR DESCRIPTION
**Pull request by meln1k**
# Description

Add support for collecting  Digital Ocean resources, Part 1.

The collector will be split into two parts due to its size

# To-Dos
Part 1
- [x] Teams (aka Accounts)
- [x] Instances
- [x] Volumes
- [x] Databases
- [x] Floating IPs
- [x] K8s
- [x] Load balancers
- [x] Projects
- [x] Project resources
- [x] Regions
- [x] Snapshots
- [x] VPCs
- [x] Support pagination
- [x] collect multiple accounts

Part 2:
- [ ] Images
- [ ] CDN endpoints
- [ ] Certificates
- [ ] Container registry
- [ ] Apps
- [ ] Use the hierarchical ids
- [ ] SSH keys 
- [ ] Tags
- [ ] Spaces
- [ ] Domain Records
- [ ] Domains
- [ ] Firewalls
- [ ] Monitoring Alert Policies

- [ ] Add test coverage for new or updated functionality
- [x] Lint and test with `tox`
- [ ] Document new or updated functionality (someengineering/resoto.com#XXXX)

# Code of Conduct

By submitting this pull request, I agree to follow the [code of conduct](https://resoto.com/code-of-conduct).

### CODE REVIEW

File: plugins/digitalocean/resoto_plugin_digitalocean/resources.py [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+    """DigitalOcean project"""
+
+    kind: ClassVar[str] = "digitalocean_project"
+
+    def delete(self, graph: Graph) -> bool:
+        return NotImplemented
+
+
+@dataclass(eq=False)
+class DigitalOceanInstance(DigitalOceanResource, BaseInstance):
```
Comment (aquamatthias) [inline plugins/digitalocean/resoto_plugin_digitalocean/resources.py:85]: Call it Droplet?

File: plugins/digitalocean/resoto_plugin_digitalocean/client.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        json_response = requests.get(url, headers=headers).json()
+        result = result + json_response.get(payload_object_name, [])
+
+        while json_response.get("links", {}).get("pages", {}).get("last", "") != url:
+            url = json_response.get("links", {}).get("pages", {}).get("next", "")
+            if url == "":
+                break
+            log.debug(f"fetching {url}")
+            json_response = requests.get(url, headers=headers).json()
+            result = result + json_response.get(payload_object_name, [])
```
Comment (aquamatthias) [inline plugins/digitalocean/resoto_plugin_digitalocean/client.py:39]: You can use `list.extend` when the list is created locally in this method.

File: plugins/digitalocean/resoto_plugin_digitalocean/collector.py [hunk 3]
```diff
@@ -0,0 +1,622 @@
...
+    def collect_vpcs(self) -> None:
+        vpcs = self.client.list_vpcs()
+        self.collect_resource(
+            vpcs,
+            resource_class=DigitalOceanNetwork,
+            attr_map={
+                "id": "urn",
+                "ip_range": "ip_range",
+                "description": "description",
+                "defalut": "defalut",
```
Comment (aquamatthias) [inline plugins/digitalocean/resoto_plugin_digitalocean/collector.py:473]: dafault?

**Review by aquamatthias — APPROVED** [review]
Would love to see (can be a separate PR):
- a (at least minimal) test coverage
- running mypy in strict mode is green.

Other than that it is a great start - thanks!

**Review by aquamatthias — APPROVED** [review]
Thanks!

**Review by TheCatLady — APPROVED** [review]

**Pull request merged**